x-pack/plugins/enterprise_search/public/plugin.ts

```diff
@@ -153,7 +156,9 @@ export class EnterpriseSearchPlugin implements Plugin {
     }
   }
 
-  public start() {}
+  public start(core: CoreStart) {
+    docLinksService.setDocLinks(core.docLinks);
+  }
```

Comment:
I would strongly prefer to do this in `public/applications/index.tsx` instead of `public/plugin.ts`. We're already passing `coreStart` to our plugins via `getKibanaDependencies` so that we don't have to do anything extra in the `start()` method.

This also helps us organizationally:

1. `public.ts` remains primarily concerned with **getting** & passing data (primarily from other Kibana dependencies, or from our own initial data API call)
2. `index.tsx` remains primarily concerned with **setting** passed data into state, logic files, or static URL helpers (this one + the external enterprise search URL)

Example diff, feel free to cherry-pick: https://github.com/constancecchen/kibana/commit/b190ff292e0138a7e235312d02b09a88063d5424